File: javascript/uniqueCharacter.js
function hasAllUniqueCharacters(str) {
    let checker = 0;
    for (let i = 0; i < str.length; i++) {
        const c = str[i];
        if (c < 'a' || c > 'z') {
            throw new Error(`Character '${c}' is not a lowercase letter (a-z).`);
        }
        const bitIndex = str.charCodeAt(i) - 'a'.charCodeAt(0);
        if ((checker & (1 << bitIndex)) !== 0) {
            return false;
        }
        checker |= (1 << bitIndex);
    }
    return true;
}

const testCases = ["abcdef", "aabcde", "xyz", "aabb", "programming"];
for (const input of testCases) {
    console.log(`"${input}" -> ${hasAllUniqueCharacters(input) ? "All unique" : "Has duplicates"}`);
}

File: kotlin/UniqueCharacter.kt
fun hasAllUniqueCharacters(str: String): Boolean {
    var checker = 0
    for (c in str) {
        require(c in 'a'..'z') { "Character '$c' is not a lowercase letter (a-z)." }
        val bitIndex = c - 'a'
        if (checker and (1 shl bitIndex) != 0) {
            return false
        }
        checker = checker or (1 shl bitIndex)
    }
    return true
}

fun main() {
    val testCases = listOf("abcdef", "aabcde", "xyz", "aabb", "programming")
    for (input in testCases) {
        println("\"$input\" -> ${if (hasAllUniqueCharacters(input)) "All unique" else "Has duplicates"}")
    }
}
